src/unix/notbsd/linux/other/mod.rs

```diff
@@ -44,11 +57,24 @@ s! {
         pub ss_size: ::size_t
     }
 
+    pub struct addr_bnd_t {
+        lower: *mut ::c_void,
+        upper: *mut ::c_void,
+    }
+
+    pub struct sigfault_t {
+        addr: *mut ::c_void,
+        #[cfg(target_arch = "sparc")]
+        trapno: ::c_int,
+        addr_lsb: ::c_short,
+        xxx_name_unclear: XXX_not_sure_what_to_call_this
+    }
+
     pub struct siginfo_t {
         pub si_signo: ::c_int,
         pub si_errno: ::c_int,
         pub si_code: ::c_int,
-        pub _pad: [::c_int; 29],
+        sifields: sifields_t,
```

Comment:
In the open issue #716 I was hoping that the padding field would be private, which would have allowed us to change this, but for some reason it is public, so this would be a hard breaking change, which I don't think we can make until the next breaking version is released.

Reply:
Depending on how long we expect it to take to get this into a mergable state, perhaps it would make sense for me to send a seperate PR to make that field private, which could be merged/released sooner?